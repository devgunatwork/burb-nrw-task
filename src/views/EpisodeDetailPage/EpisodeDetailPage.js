import React from 'react';
import { useParams } from 'react-router-dom';
import Typography from '@material-ui/core/Typography';
import Details from '../../components/Details';
import SplitPageLayout from '../../components/SplitPageLayout';
import checkImageUrl from '../../utils/checkImageUrl';
import { episodeDetailPageSampleResponse } from '../ShowPage/sampleResponse';

/**
 * @description component is container for episode detail page
 * */
const EpisodeDetailPage = () => {
  const { episodeId } = useParams();
  const { image, name, summary } = episodeDetailPageSampleResponse;
  const url = checkImageUrl(image);

  return (
    <SplitPageLayout imageUrl={url}>
      <Typography variant="h5">EPISODE DETAILS</Typography>
      <Details name={name} summary={summary} id={episodeId} />
    </SplitPageLayout>
  );
};

export default EpisodeDetailPage;
