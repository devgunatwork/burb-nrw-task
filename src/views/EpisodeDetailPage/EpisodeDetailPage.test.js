import React from 'react';
import EpisodeDetailPage from './EpisodeDetailPage';

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useParams: () => ({
    id: '1234',
  }),
}));

describe('views', () => {
  describe('EpisodeDetailPage', () => {
    describe('UI', () => {
      it('should match the snapshot', () => {
        const wrapper = shallow(<EpisodeDetailPage />);

        expect(wrapper).toMatchSnapshot();
      });
    });
  });
});
