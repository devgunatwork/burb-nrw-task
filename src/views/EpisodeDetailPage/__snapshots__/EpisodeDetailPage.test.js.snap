// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`views EpisodeDetailPage UI should match the snapshot 1`] = `
<SplitPageLayout
  imageUrl="http://static.tvmaze.com/uploads/images/original_untouched/127/319448.jpg"
>
  <WithStyles(ForwardRef(Typography))
    variant="h5"
  >
    EPISODE DETAILS
  </WithStyles(ForwardRef(Typography))>
  <Details
    name="Sideline Dad"
    summary="<p>When the Girls start playing soccer, the Professor goes overboard.</p>"
  />
</SplitPageLayout>
`;
